frontend/src/components/Dashboard/SideBar.js

```diff
@@ -0,0 +1,261 @@
+                    if (stateIndex < states.length) {
+                        bundlesDict[states[stateIndex]] = data.response.result;
+                        fetchBundles(stateIndex + 1, bundlesDict);
+                    } else {
+                        bundlesDict['floating'] = data.response.result;
+                        const bundles: HTMLElement[] = [];
+                        for (let state in bundlesDict) {
+                            // Only show the non-zero items
+                            if (bundlesDict[state] > 0) {
+                                bundles.push(
+                                    <li key={state} className={classes.subheader}>
+                                        {state + ': ' + bundlesDict[state]}
+                                    </li>,
+                                );
+                            }
+                        }
+                        this.setState({ bundles: bundles });
+                    }
+                },
+                error: (xhr, status, err) => {
+                    console.error(xhr.responseText);
+                },
+            });
+        };
+        // Start to fetch the bundles' count
+        fetchBundles(0, {});
+    }
+
+    render() {
+        const { classes, userInfo } = this.props;
+        if (!userInfo) {
+            return null;
+        }
+        return (
+            <Card className={cx(classes.card)} elevation={0} style={{ height: '100%' }}>
+                <Avatar src={'https://i.pravatar.cc/30'} className={classes.avatar} />
+                <Box className={classes.box}>
+                    <h3 className={classes.name}>{userInfo.user_name}</h3>
+                    {userInfo.affiliation ? (
+                        <p className={classes.affiliation}>Affiliation: {userInfo.affiliation}</p>
+                    ) : null}
+                </Box>
+                <Divider />
+
+                {this.props.showQuota ? (
+                    <Box className={classes.box}>
+                        <a
+                            className={classes.subheader}
+                            href={
+                                'https://github.com/codalab/codalab-worksheets/blob/master/docs/FAQ.md#how-do-i-reduce-the-amount-of-disk-usage'
+                            }
+                        >
+                            Disk Usage (Bytes)
+                        </a>
+                        <Box className={classes.progressBox}>
+                            <BorderLinearProgress
+                                className={classes.margin}
+                                variant='determinate'
+                                color='secondary'
+                                value={(userInfo.disk_used / userInfo.disk_quota) * 100}
+                            />
+                            <span className={classes.value}>
+                                {Math.floor(userInfo.disk_quota / 1024 / 1024) + 'GB'}
```

Comment:
Can you use the general function for rendering sizes that we use elsewhere?

Reply:
Thanks for the advice! I used the general function instead.